js/summary.js

```diff
@@ -116,12 +128,18 @@ H5P.Summary = (function ($, Question, JoubelUI) {
 
     // Register retry button to h5p question
     var self = this;
-    this.addButton('try-again', 'retry', function () {
 
-      // Remove button after is has been clicked
-      self.hideButton('try-again');
-      self.resetTask();
-    }, false);
+    // Attach the retry buton if it is enabled in the semantics
+    if (this.options.behaviour.enableRetry) {
+
+      // Add JoubelUI retry button using the h5p-question module
+      this.addButton('try-again', 'retry', function () {
+
+        // Remove button after is has been clicked
+        self.hideButton('try-again');
+        self.resetTask();
+      }, self.progress == self.summaries.length);
```

Comment:
Two equals enables type coercion in the equality calculation. It is fragile in this comparison since undefined, false, 0, etc. can evaluate to false, thus if progress is 0 and summaries length is undefined for some reason this logic will not hold. You usually want to use '===' for equality comparison, which will disallow coercion in the comparison, unless you explicitly want to use type coercion.